src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -133,7 +133,7 @@ internal static bool TryCreate(SyntheticBoundNodeFactory F, MethodSymbol method,
                     createBuilderMethod: createBuilderMethod,
                     taskProperty: null,
                     setException: null, // unused
-                    setResult: WellKnownMember.System_Runtime_CompilerServices_AsyncIteratorMethodBuilder__Complete,
+                    setResult: WellKnownMember.System_Runtime_CompilerServices_AsyncIteratorMethodBuilder__Complete, // AsyncIteratorMethodBuilder.Complete is the corresponding method to AsyncTaskMethodBuilder.SetResult
```

Comment:
I find this kind of a confusing pattern, but I'm not sure how I would change it

Reply:
The BCL initially had AsyncIteratorMethodBuilder.SetResult (like other builders), but then changed the name since there is no value/result. Still is it the spiritual equivalent... ;-)

---
In reply to: [337678803](https://github.com/dotnet/roslyn/pull/39436#discussion_r337678803) [](ancestors = 337678803)